server/parsec/components/memory/invite.py

```diff
@@ -257,7 +258,12 @@ async def _conduit_listen(
                 invitation.conduit_state == NEXT_CONDUIT_STATE[ctx.state]
                 and curr_our_payload is None
             ):
-                return ctx.peer_payload, invitation.conduit_is_last_exchange
+                # Careful here: it's possible that peer has already changed the state
```

Comment:
:clap: :1st_place_medal:

Reply:
> Shouldn't we also change the line 250 given it also returns invitation.conduit_is_last_exchange ?

No, this would return the wrong result.

>  (if not, we should add a comment)

I'll try to write something.

Reply:
[195d06b](https://github.com/Scille/parsec-cloud/pull/7307/commits/195d06bc98e169d4a6f8583ab5f38c1bd1ef9de6)